src/components/DatePicker/index.ios.js

```diff
@@ -36,9 +39,20 @@ class DatePicker extends React.Component {
      * @param {Event} event
      */
     showPicker(event) {
-        Keyboard.dismiss();
+        /**
+         * Ios will auto dismiss the keyboard when popover is opened and open again when it's closed
+         * We need the keyboardDidHide listener to make sure that the popover will only be opened after the keyboard is closed
+         */
+        if (this.props.isKeyboardShown) {
+            const listener = Keyboard.addListener('keyboardDidHide', () => {
+                this.setState({isPickerVisible: true});
+                listener.remove();
+            });
+            Keyboard.dismiss();
+        } else {
+            this.setState({isPickerVisible: true});
+        }
```

Comment:
We prefer the use of early returns, putting the shorter condition block first is easier to read, this also better matches the structure of the comment I suggested above

```suggestion
        if (!this.props.isKeyboardShown) {
            this.setState({isPickerVisible: true});
            return;
        } 
        const listener = Keyboard.addListener('keyboardDidHide', () => {
            this.setState({isPickerVisible: true});
            listener.remove();
        });
        Keyboard.dismiss();
```